fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreIndexTest.java

```diff
@@ -2354,8 +2362,12 @@ public void boundaryPrimaryKeys() throws Exception {
 
     @Test
     public void noBoundaryPrimaryKeys() throws Exception {
+        final FDBDatabaseFactory factory = FDBDatabaseFactory.instance();
+        factory.setLocalityProvider(MockedLocalityUtil.instance());
+        FDBDatabase database = FDBDatabaseFactory.instance().getDatabase();
```

Comment:
I think you *probably* have a problem here.  The `FDBDatabaseFactory` maintains a cache of the database instances, so if another test has already created the `FDBDatabase` then you will be using the `FDBLocalityProvider` that was previously installed.  Or, if you do install the `MockedLocalityUtil` provider, the next test that runs and needs a locality provider will end up using the one you just installed.

It would probably make sense to add a `@Before` method that does: `FDBDatabaseFactory.instance().clear()`. This will at least remove the cached database instances, but it won't take care of the locality API that you've installed.  I guess that could be handled in the test cases themselves for now.

 Or, optional, in the JUnit4 world, we could have had a:

    @Rule
    FDBDatabaseFactoryRule databaseFactory = new FDBDatabaseFactoryRule();
    ...
    FDBDatabase database = databaseFactory.getDatabase();

that would automatically reset the cached databases before each test...it appears that the way to do this in JUnit5 is with `@Extension`, but that might be a bit much for this PR (although, it would generally be very useful).   I think I'll open a separate issue for this.

Reply:
Yeah, that's right! In fact, I used to call `clear()` but later when @nblintao asked about the reason, I didn't recall and dropped it as the tests passed without it. Now I recall, I had hit the exact same problem. I'll fix the immediate problem with the easy way, and open an issue (I've never used the `@Extension`  before - so I'll defer it but will get back to it soon). Thanks for the review.

Reply:
It should be fixed now. I think you opened an issue for this already (#591). thanks.